app/src/main/java/in/co/rajkumaar/amritarepo/activities/LaunchingActivity.java

```diff
@@ -759,5 +694,41 @@ private void popupSnackbarForCompleteUpdate() {
                 getResources().getColor(R.color.white));
         snackbar.show();
     }
+
+    private void startUpdate() {
+        appUpdateInfoTask.addOnSuccessListener(appUpdateInfo -> {
+            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE && appUpdateInfo.clientVersionStalenessDays() != null) {
```

Comment:
This means if stale days is null, then the update won’t even be shown which isn’t ideal. Instead we can show a flexible update if the stale days is null. Also as I already mentioned, the if and else block have more or less same code, the only changing part is the update type which you can pass using conditional operator.